futures-core/src/task/mod.rs

```diff
@@ -1,35 +1,142 @@
+//! Tasks used to drive a future computation
+//!
+//! It's intended over time a particular operation (such as servicing an HTTP
+//! request) will involve many futures. This entire operation, however, can be
+//! thought of as one unit, as the entire result is essentially just moving
+//! through one large state machine.
+//!
+//! A "task" is the unit of abstraction for what is driving this state machine
```

Comment:
@aturon Is this still how we want to talk about "tasks"? My intuition always lead me to think that a "task" was the top-level future driven by an executor -- we'd even talked about `trait Task = Future<Item = (), Error = ()>;`.

Reply:
Oh right yeah so I've just copied/pasted doc stuff around, I'm thinking we'll want docs to be a future PR. Right now it's all likely horribly outdated.